### DISCUSSION TITLE
multiple yaml docs to a json array

### DISCUSSION THREAD

**Discussion by coderplay** [mikefarah/yq#993] [Q&A]
I got a single yaml file with multple doc

```
a: 1
---
b: 2
```

How can I output a json like this
```
[
  {
    "a": 1
  },
  {
    "b": 2
  }
]
```

When I use `yq e  sample.yaml -j` , it generates a invalid json
```
{
  "a": 1
}
{
  "b": 2
}
```

**Comment by mikefarah** [comment]
The command you've got up there creates a json doc for each yaml doc - which is why you are seeing that output.

What you'll need to do is:

```
yq ea '[.]' sample.yaml -j
```

- `ea` (or eval-all) to read all the documents at once (default is to read one doc at a time)
- `[.]` to put all the documents in an array

**Comment by coderplay** [comment]
@mikefarah  Thanks for your quick reply. This command can't read the yaml from `stdout`

```
% cat /tmp/sample | yq ea -j
{
  "a": 1
}
{
  "b": 2
}
```

if I add '[.]'

```
 % cat /tmp/sample | yq ea '[.]' -j
Error: open [.]: no such file or directory
```

**Reply by mikefarah** [reply]
You just need to give it a `-` so it knows to read from STDIN:

cat file.yaml | yq e '.a.b[0].c' -
